### ISSUE TITLE
Edits should be saved as a draft

### ISSUE THREAD

**Issue by jakerr** [markusfisch/ShaderEditor#68]
Backgrounding the app for any reason such as responding to a chat, or accidently pressing home causes the shader in progress to be reverted to the last save.

I'm often loosing tens of minutes of coding time because I forget to save before backgrounding.
I don't tend to manually run the shader, so the "Save on run" setting doesn't help me.

I would prefer if edits were always saved as a draft. There could be a button to revert to the last manual save so that reverting is always an explicit action rather than a surprise.

**Comment by markusfisch** [comment]
Yes, that bit me too ... I'd like to implement undo/redo for the editor and always save after a change. It's on the list.

**Comment by claudeha** [comment]
https://github.com/markusfisch/ShaderEditor/commit/9bd232c4195dc4de70e3f9a6c7a94634c0400396 adds autosave but warns of no undo so it is disabled by default

https://github.com/markusfisch/ShaderEditor/commit/cd69a0db6db6a2f9d23cf2e8dfd56532316c40c7 adds undo and redo

Perhaps the autosave can be enabled by default now?

**Comment by markusfisch** [comment]
Enabled by default in https://github.com/markusfisch/ShaderEditor/commit/f1f8eba9a89953679d25117462ec6832ef985c74

Will be in the next version 😄

**Comment by AntonPieper** [comment]
I think this issue can be closed, or is there anything left to do?

**Comment by markusfisch** [comment]
Yes, closed 👍
